eNMS/services/file_transfer/generic_file_transfer.py

```diff
@@ -61,7 +61,7 @@ def job(self, run, payload, device):
             else:
                 pairs = []
                 for glob_source in glob_source_file_list:
-                    if Path(glob_source).is_dir():
+                    if run.direction.lower() == "put" and Path(glob_source).is_dir():
```

Comment:
in the form, it says glob pattern only works with "put", so if we need to check for the direction it's not here but before that